greengrassTestSuite.py

```diff
@@ -153,21 +202,23 @@ def _check_greengrass_group_deployment_status(self, deployment_id):
 
     def create_deployment(self,
                           thingArn,
-                          component_name,
-                          component_version,
+                          component_list,
                           deployment_name="UATinPython"):
+        component_parsed_dict = {}
+        for component in component_list:
+            component_parsed_dict[component[0]] = {
+                "componentVersion": component[1]
+            }
+
         result = self._ggClient.create_deployment(
             targetArn=thingArn,
             deploymentName=deployment_name,
-            components={
-                component_name: {
-                    "componentVersion": component_version
-                }
-            },
+            components=component_parsed_dict,
         )
 
         if result is not None:
-            self._ggServiceList.append(component_name)
+            for component in component_list:
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

To create a `list`, try to use `list` comprehension instead of a loop. List comprehension is the preferred way to make a list using Python, and it&#39;s simpler and easier to understand than using a loop.

[Learn more](https://docs.python.org/3/tutorial/datastructures.html#list-comprehensions)